src/content/developers/docs/smart-contracts/formal-verification/index.md

```diff
@@ -0,0 +1,136 @@
+
+### 2. False negatives {#false-negatives}
+
+Formal verification can only check if the execution of the smart contract matches formal specifications. If specifications are poorly written, certain vulnerabilities cannot be detected by the formal verification audit. In this case, a developer might erroneously assume that the contract is bug-free.
+
+## Use formal verification tools {#use-formal-verification-tools}
+
+### Creating formal specifications {#creating-formal-specifications}
+
+**Act - _High-level specification language for defining properties for EVM smart contracts._**
+
+- [GitHub](https://github.com/ethereum/act)
+- [Documentation](https://ethereum.github.io/act/)
+
+**Scribble -** **_A specification language tool for translating high-level specifications into Solidity code._**
+
+- [Website](https://consensys.net/diligence/scribble/)
+- [Documentation](https://docs.scribble.codes/language/introduction)
+
+### Fuzzing smart contracts {#fuzzing-smart-contracts}
+
+**Echidna - _A fast contract fuzzer for detecting vulnerabilities in smart contracts through property-based testing._**
+
+- [GitHub](https://github.com/crytic/echidna/)
+
+**Harvey - _Automated fuzzing tool useful for detecting property violations in smart contract code._**
+
+- [Website](https://consensys.net/diligence/fuzzing/)
+
+### Code analysis {#code-analysis}
+
+**Oyente - _Static analysis tool for analyzing vulnerabilities in EVM bytecode with symbolic execution._**
+
+- [GitHub](https://github.com/melonproject/oyente)
+
+**Mythril - _EVM bytecode assessment tool for detecting contract vulnerabilities using taint analysis, concolic analysis, and control flow checking._**
+
+- [GitHub](https://github.com/ConsenSys/mythril-classic)
+- [Documentation](https://mythril-classic.readthedocs.io/en/master/about.html)
+
+**Manticore _- A tool for analyzing EVM bytecode analysis tool based on symbolic execution._**
+
+- [GitHub](https://github.com/trailofbits/manticore)
+- [Documentation](https://github.com/trailofbits/manticore/wiki)
+
+**Securify** **_- A security analyzer for analyzing Solidity source code in Ethereum smart contracts and detecting property violations._**
+- [GitHub](https://github.com/eth-sri/securify2)
+
+### Formal verification projects in Ethereum {#formal-verification-projects-in-ethereum}
+**hevm - _An implementation of the Ethereum virtual machine (EVM) made specifically for symbolic execution, unit testing and debugging of smart contracts._**
+- [GitHub](https://github.com/dapphub/dapptools/tree/master/src/hevm)
+
+**KEVM - _A correct-by-construction implementation of the Ethereum Virtual Machine (EVM) specified in the K Framework for implementing formal verification._**
+- [GitHub](https://github.com/runtimeverification/evm-semantics)
+- [Documentation](https://jellopaper.org/)
```

Comment:
There are quite a few common tools missing in this list. If the goal is to list only the "most used" or "most stable" ones, I'd have a few suggestions of tools to remove and to add.

Reply:
@leonardoalt what tools would you recommend adding? I'd default to you on which to keep and/or remove.

Also happy to default to you on ordering... if it makes sense to put the most used tools at the top for visibility then let's do that.

Reply:
The idea was mostly to highlight most-used and maintained tools. For instance, MAIAN hasn't been maintained for a while and the KLab repo has been archived (I took most of the tools from your listing). That said, any suggestions on what tools to add or remove are welcome.